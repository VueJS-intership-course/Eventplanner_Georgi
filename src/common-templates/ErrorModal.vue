<template>
    <div class="modal" id="exampleModalCenter" >
        <div class="modal-dialog modal-dialog-centered" >
            <div class="modal-content">
                <div class="modal-header">
                    <h5 class="modal-title text-" id="exampleModalCenterTitle">Error occured</h5>
                    <button type="button" class="close">
                        <span >&times;</span>
                    </button>
                </div>
                <div class="modal-body">
                    ...
                </div>
                <div class="modal-footer">
                    <button type="button" class="btn btn-secondary" >Close</button>
                    <button type="button" class="btn btn-primary">Save changes</button>
                </div>
            </div>
        </div>
    </div>
</template>