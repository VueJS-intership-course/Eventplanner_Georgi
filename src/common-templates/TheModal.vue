<template>
    <div class="back-modal">
        <div class="add-modal container py-3 d-flex justify-content-center">
               <slot></slot>
        </div>
    </div>
</template>

<script setup>
import { onBeforeMount, onBeforeUnmount } from 'vue';

onBeforeMount(() => {
    document.body.style.overflow = 'hidden';
})

onBeforeUnmount(() => {
    document.body.style.overflow = '';
})
</script>

<style lang="scss" scoped>

.back-modal {
    @include modal-style;
};
</style>