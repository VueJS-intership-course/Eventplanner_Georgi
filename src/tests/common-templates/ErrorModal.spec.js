import ErrorModal from '@/common-templates/ErrorModal.vue';
import { mount, shallowMount } from '@vue/test-utils';
import { it, expect } from '@jest/globals';
import TheModal from '@/common-templates/TheModal.vue'

const wrapper = mount(ErrorModal, {
    props: {
        errorMsg: 'Error message',
    },
    global: {
        components: {
            'TheModal': TheModal 
        }
    }
});

it('should show the error message', async () => {
    expect(wrapper.find('.modal-body').text()).toBe('Error message');
});

it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
});

