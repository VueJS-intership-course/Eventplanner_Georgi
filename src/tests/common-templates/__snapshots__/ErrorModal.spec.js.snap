// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div class="back-modal">
  <div class="add-modal container py-3 d-flex justify-content-center">
    <div class="error-modal bg-light p-4" id="exampleModalCenter">
      <div class="modal-dialog modal-dialog-centered">
        <div class="modal-content">
          <div class="modal-header mb-2 p-3">
            <h5 class="modal-title text-danger fw-bold" id="exampleModalCenterTitle">Error occured</h5><button class="btn btn-primary"><span>Ã—</span></button>
          </div>
          <div class="modal-body mb-3 p-3">Error message</div>
          <div class="modal-footer p-3"><button type="button" class="btn btn-outline-primary">Close</button></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
